### PR TITLE
Implement Classroom, Book, Rental Classes, and Relationships

### PR DESCRIPTION
**Pull request by axcell-0**
** Description:**
This pull request addresses the project requirements by introducing and modifying classes and establishing relationships as follows:

Classroom Class: Implemented the Classroom class with an @label instance variable, created getter and setter methods for @label, and established a has-many/belongs-to relationship with the Student class. Adding a student to a classroom now automatically sets the classroom for the student, ensuring consistency.

Book Class: The Book class is now in place with @title and @author instance variables, along with getters and setters for these attributes. Additionally, have added support for a has-many relationship with Rental instances, allowing a book to track its rentals.

Rental Class: The Rental class has been implemented with an @date instance variable and associated getters and setters.have established belongs-to relationships between Rental and both Book and Person. Moreover, the Rental constructor now sets the Book and Person attributes, enhancing the connections between these entities.

Person Class: The Person class has been modified to facilitate has-many and belongs-to relationships with Rental instances. This means that a person can now have multiple rentals, and each rental correctly belongs to a person.

Book-Rental Relationship Enhancement: enhanced the relationship between Book and Rental. The add_rental method in the Book class now sets the book for the rental, completing the many-to-many relationship with Person through Rental.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :title, :author, :rentals
+
+  def initialize(label)
+    @label = label
```
Comment (rloterh) [inline classroom.rb:5]: - [ ] As in this app we intend to both _read_ and _modify_ `@label` let's ensure to mark it with `attr_accessor`, like below:
```ruby
class Classroom
  attr_accessor :label 
 
  def initialize(label)
    @label = label
    .
  end
end
```

Let's endeavor to read further on the same [here](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#useful-shortcuts-attr_reader-attr_writer-and-attr_accessor`)

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(book, person, _date)
+    @book = book
+    @person = person
+    book.rentals << self
+    person.rentals << self
+  end
```
Comment (rloterh) [inline rental.rb]: - [ ] Let's ensure to initialize `@date` instance in the `Rental` constructor so as to set its initial state when a new Rental is created, like below: 

```ruby
class Rental
  attr_accessor :date
  
  def initialize(date)
    @date = date
  end
end
```

Let's endeavor to read further on the same [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars)

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Axcell, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Properly established 'belongs-to' relationship!
- Nice job so far setting up associations!
-  ReadMe looks professional!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @axcell-0, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**